code/game/objects/structures/crates_lockers/closets/gimmick.dm

```diff
@@ -113,13 +113,29 @@
 
 /obj/structure/closet/mini_fridge
 	name = "grimy mini-fridge"
-	desc = "A small contraption designed to imbue a few drinks with a pleasant chill. This antiquated unit however seems to serve no purpose other than keeping the roaches company."
+	desc = "A small contraption designed to imbue a few drinks with a pleasant chill."
 	icon_state = "mini_fridge"
 	icon_welded = "welded_small"
 	max_mob_size = MOB_SIZE_SMALL
+	pass_flags = PASSTABLE
+	anchored_tabletop_offset = 3
+	anchored = 1
 	storage_capacity = 10
 
 /obj/structure/closet/mini_fridge/PopulateContents()
+	. = ..()
+	new /obj/effect/spawner/random/food_or_drink/refreshing_beverage(src)
+	new /obj/effect/spawner/random/food_or_drink/refreshing_beverage(src)
+	if(prob(50))
+		new /obj/effect/spawner/random/food_or_drink/refreshing_beverage(src)
+	if(prob(40))
+		new /obj/item/reagent_containers/cup/glass/bottle/beer(src)
+
+/obj/structure/closet/mini_fridge/grimy
+	name = "grimy mini_fridge"
+	desc = "A small contraption designed to imbue a few drinks with a pleasant chill. This antiquated unit however seems to serve no purpose other than keeping the roaches company."
+
+/obj/structure/closet/mini_fridge/grimy/PopulateContents()
 	. = ..()
```

Comment:
this means that grimy mini fridges will start with at least four refreshing beverages at the start since we're invoking parents, not to mention the possibility for beer or another refereshing beverage. just cut out lines 140 to 143 and we should be okay

Reply:
Oh so the parent Populate Contents() Gets applied to the child too? alright.